PyNormaliz.py

```diff
@@ -242,340 +242,75 @@ def PrintHilbertSeries(self):
-        return PyNormaliz_cpp.NmzResult(self.cone, "VirtualMultiplicity")
-
-    def IsGorenstein(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("IsGorenstein")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "IsGorenstein")
-
-    def GeneratorOfInterior(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("GeneratorOfInterior")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "GeneratorOfInterior")
-
-    def VerticesFloat(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("VerticesFloat")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "VerticesFloat")
-
-    def Volume(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("Volume")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "Volume")
-
-    def SuppHypsFloat(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("SuppHypsFloat")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "SuppHypsFloat")
-
-    def LatticePoints(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("LatticePoints")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "LatticePoints")
-
-    def EuclideanIntegral(self, **kwargs):
-        input_list = self.__process_keyword_args(kwargs)
-        input_list.append("EuclideanIntegral")
-        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
-        return PyNormaliz_cpp.NmzResult(self.cone, "EuclideanIntegral")
+    def _generic_getter(self, name, **kwargs):
+        input_list = self.__process_keyword_args(kwargs)
+        input_list.append(name)
+        PyNormaliz_cpp.NmzCompute(self.cone, input_list)
+        return PyNormaliz_cpp.NmzResult(self.cone, name)
+
+
+# Generate getters for a bunch of Normalize properties
+def add_dyn_getter(name):
+    def inner(self, **kwargs):
+        return self._generic_getter(name, **kwargs)
+    inner.__doc__ = "docstring for %s" % name
+    inner.__name__ = name
+    setattr(Cone, name, inner)
+
+add_dyn_getter("AffineDim")
+add_dyn_getter("ClassGroup")
+add_dyn_getter("ConeDecomposition")
+add_dyn_getter("Congruences")
+add_dyn_getter("Deg1Elements")
+add_dyn_getter("Dehomogenization")
+add_dyn_getter("EhrhartQuasiPolynomial")
+add_dyn_getter("EmbeddingDim")
+add_dyn_getter("Equations")
+add_dyn_getter("EuclideanIntegral")
+add_dyn_getter("ExcludedFaces")
+add_dyn_getter("ExternalIndex")
+add_dyn_getter("ExtremeRays")
+add_dyn_getter("FVector")
```

Comment:
Instead of hardcoding the list of wrappers, we could also generate it during runtime, by adding a new function to the C++ code which returns a list of strings with all known property names. Then, we just call that function here; iterate over the list it returns; and call `add_dyn_getter` on each.
To avoid clashes with those attributes like `HilbertSeries` that already have a custom getter, we can modify `add_dyn_getter` to not do anything if a method with the given name already exists.